src/blockMonitor/blockProcessor.ts

```diff
@@ -109,27 +121,28 @@ export class BlockProcessor<T extends IBlockStub> extends StartStopService {
         return this.lastBlockHashReceived === blockHash;
     }
 
-    // update the new headHash if needed, and emit the appropriate events
-    private processNewHead(headBlock: T, commonAncestorBlock: T | null) {
+    // update the new headHash and emit the appropriate events
+    private processNewHead(headBlock: Readonly<T>, commonAncestorBlock: T | null) {
         const oldHeadHash = this.headHash; // we need to remember the old head for proper Reorg event handling
         this.headHash = headBlock.hash;
 
-        if (this.isBlockHashLastReceived(this.headHash)) {
-            this.mBlockCache.setHead(headBlock.hash);
-
-            // Emit the appropriate events, but only if the service is already started
-            if (this.started) {
-                if (!commonAncestorBlock) {
-                    // reorg beyond the depth of the cache; no common ancestor found
-                    this.emit(BlockProcessor.REORG_EVENT, null, this.headHash, oldHeadHash);
-                } else if (oldHeadHash !== commonAncestorBlock.hash) {
-                    // reorg with a known common ancestor in cache
-                    this.emit(BlockProcessor.REORG_EVENT, commonAncestorBlock.hash, this.headHash, oldHeadHash);
-                }
+        this.mBlockCache.setHead(headBlock.hash);
 
-                this.emit(BlockProcessor.NEW_HEAD_EVENT, headBlock.number, headBlock.hash);
-            }
+        // Emit the appropriate events, but only if the service is already started
+        if (!commonAncestorBlock) {
+            // reorg beyond the depth of the cache; no common ancestor found
+            this.emit(BlockProcessor.REORG_EVENT, null, this.headHash, oldHeadHash);
+        } else if (oldHeadHash !== commonAncestorBlock.hash) {
+            // reorg with a known common ancestor in cache
+            this.emit(BlockProcessor.REORG_EVENT, commonAncestorBlock.hash, this.headHash, oldHeadHash);
         }
```

Comment:
We are actually not using REORG_EVENTs anywhere, now; we might want to remove them.

Reply:
Agreed

Reply:
Done